src/js/stores/MesosStore.js

```diff
@@ -0,0 +1,221 @@
+var EventEmitter = require("events").EventEmitter;
+
+var AppDispatcher = require("../AppDispatcher");
+var InfoStore = require("../stores/InfoStore");
+var InfoActions = require("../actions/InfoActions");
+var InfoEvents = require("../events/InfoEvents");
+var MesosActions = require("../actions/MesosActions");
+var MesosEvents = require("../events/MesosEvents");
+
+const FILES_TTL = 60000;
+const STATE_TTL = 60000;
+const MASTER_ID = "master";
+
+var stateMap = {};
+var taskFileMap = {};
+var taskFileRequestQue = [];
+
+var MesosStore = Object.assign({
+
+  getState: function (nodeId) {
+    var data = stateMap[nodeId];
+    if (data == null || data.timestamp + STATE_TTL < Date.now()) {
+      invalidateState(nodeId);
+      return null;
+    }
+    return data.state;
+  },
+
+  getTaskFiles: function (taskId) {
+    var data = taskFileMap[taskId];
+    if (data == null || data.timestamp + FILES_TTL < Date.now()) {
+      invalidateTaskFiles(taskId);
+      return null;
+    }
+    return data.files;
+  }
+
+}, EventEmitter.prototype);
+
+function addState(nodeId, state) {
+  stateMap[nodeId] = {
+    state: state,
+    timestamp: Date.now()
+  };
+  MesosStore.emit(MesosEvents.CHANGE);
```

Comment:
Does it makes sense to split the CHANGE event into two?
This here could be ```MesosEvents.CHANGE_STATE``` and below it could be ```MesosEvents.CHANGE_TASKFILES``` ?

Then you could probably write the components even cleaner.